atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -317,6 +317,11 @@ private void checkNamespaceConfig() {
                                 + " Please contact AtlasDB support to remediate this. Specific steps are required;"
                                 + " DO NOT ATTEMPT TO FIX THIS YOURSELF.");
             }
+        } else if (timelock().isPresent()) {
+            // Special case - empty timelock and empty namespace/keyspace does not make sense
+            boolean timelockClientNonEmpty = !timelock().get().client().orElse("").isEmpty();
+            Preconditions.checkState(timelockClientNonEmpty,
+                    "For InMemoryKVS, the TimeLock client should not be empty");
```

Comment:
I think we could rewrite this as:
```
if (kvs is InMemory) {
    (namespacePresent && namespaceEqualsClientAndKeyspace) || timelockIsNotEmpty
} else {
    (namespacePresent && namespaceEqualsClientAndKeyspace) || (keyspacePresent && keyspaceEqualsClient)
}
```

Feel free to improve it even more. My goal here was to make it as descriptive as possible.